omigami/feature_evaluator.py

```diff
@@ -25,11 +37,45 @@ def __init__(
         self._n_initial_features = 0
 
     def set_n_initial_features(self, n_initial_features: int):
+        """
+        Saves the initial number of features in an internal variable to be used to
```

Comment:
The first sentence is implementation specific. Focus on the highlevel part. What does it stand for? Why does it need to be updated after instantiation? I still don't quite get what this is used for and why n_initial_features can't be set by the cosntructor (not saying that you should mentions this in the doc here).